### PR TITLE
django 1.10 compatibility

### PR DESCRIPTION
**Pull request by mgeist**
@BetterWorks/goals 
@terite

### CODE REVIEW

File: djangosaml2/tests/urls.py [hunk 1]
```diff
@@ -12,25 +12,30 @@
...
-urlpatterns = patterns(
-    'djangosaml2.views',
-    url(r'^login/$', 'login', name='saml2_login'),
-    url(r'^acs/$', 'assertion_consumer_service', name='saml2_acs'),
-    url(r'^logout/$', 'logout', name='saml2_logout'),
-    url(r'^ls/$', 'logout_service', name='saml2_ls'),
-    url(r'^metadata/$', 'metadata', name='saml2_metadata'),
-
+urls = [
+    url(r'^login/$', 'djangosaml2.views.login', name='saml2_login'),
```
Comment (terite) [inline djangosaml2/tests/urls.py]: from the other review

> Support for string view arguments to url() is removed.

File: djangosaml2/tests/urls.py [hunk 2]
```diff
@@ -12,25 +12,30 @@
...
 
+import django
+from django.contrib import admin
+
 try:
-    from django.conf.urls import patterns, include, url
+    from django.conf.urls import include, url
+    if django.VERSION < (1, 10):
+        from django.conf.urls import patterns
 # Fallback for Django versions < 1.4
```
Comment (terite) [inline djangosaml2/tests/urls.py]: If you want, I think removing 1.4 support is fine for code simplification

File: djangosaml2/tests/urls.py [hunk 3]
```diff
@@ -12,25 +12,30 @@
...
+    url(r'^logout/$', 'djangosaml2.views.logout', name='saml2_logout'),
+    url(r'^ls/$', 'djangosaml2.views.logout_service', name='saml2_ls'),
+    url(r'^metadata/$', 'djangosaml2.views.metadata', name='saml2_metadata'),
     # this is needed for the tests
     url(r'^admin/', include(admin.site.urls)),
-)
+]
+
+if django.VERSION < (1, 10):
+    urlpatterns = patterns('', *urls)
```
Comment (terite) [inline djangosaml2/tests/urls.py]: same nit about not having `urls` as an intermediate value

File: djangosaml2/tests/urls.py [hunk 4]
```diff
@@ -12,25 +12,30 @@
...
-    url(r'^logout/$', 'logout', name='saml2_logout'),
-    url(r'^ls/$', 'logout_service', name='saml2_ls'),
-    url(r'^metadata/$', 'metadata', name='saml2_metadata'),
-
+urls = [
+    url(r'^login/$', 'djangosaml2.views.login', name='saml2_login'),
+    url(r'^acs/$', 'djangosaml2.views.assertion_consumer_service', name='saml2_acs'),
+    url(r'^logout/$', 'djangosaml2.views.logout', name='saml2_logout'),
+    url(r'^ls/$', 'djangosaml2.views.logout_service', name='saml2_ls'),
+    url(r'^metadata/$', 'djangosaml2.views.metadata', name='saml2_metadata'),
```
Comment (NullSoldier) [inline djangosaml2/tests/urls.py]: You should align all of these on the comma

File: djangosaml2/urls.py [hunk 5]
```diff
@@ -13,19 +13,29 @@
...
-    url(r'^ls/$', 'logout_service', name='saml2_ls'),
-    url(r'^metadata/$', 'metadata', name='saml2_metadata'),
-)
+
+urls = [
+    url(r'^login/$', login, name='saml2_login'),
+    url(r'^acs/$', assertion_consumer_service, name='saml2_acs'),
+    url(r'^logout/$', logout, name='saml2_logout'),
+    url(r'^ls/$', logout_service, name='saml2_ls'),
+    url(r'^metadata/$', metadata, name='saml2_metadata'),
```
Comment (NullSoldier) [inline djangosaml2/urls.py]: NIT: align

File: tests/settings.py [hunk 6]
```diff
@@ -74,19 +73,12 @@
 STATICFILES_FINDERS = (
     'django.contrib.staticfiles.finders.FileSystemFinder',
     'django.contrib.staticfiles.finders.AppDirectoriesFinder',
-#    'django.contrib.staticfiles.finders.DefaultStorageFinder',
+    # 'django.contrib.staticfiles.finders.DefaultStorageFinder',
```
Comment (terite) [inline tests/settings.py]: :(

Comment (NullSoldier) [inline tests/settings.py]: NIT: commented out code

File: djangosaml2/tests/urls.py (hunk 4 above)
Comment (terite) [inline djangosaml2/tests/urls.py]: get outta here troll, we don't align nothin' on forks

Comment (mgeist) [inline djangosaml2/tests/urls.py]: a wild null appears!

File: tests/settings.py (hunk 6 above)
Comment (clarkduvall) [inline tests/settings.py]: +1 comments help code readability

**Comment by terite** [comment]
Verified that this works, lgtm

saml-works

**Comment by danielcmartin** [comment]
lgtm

**Pull request merged**